### DISCUSSION TITLE
Delay refresh every page for a few miliseconds to completely avoid ghosting

### DISCUSSION THREAD

**Discussion by AryaTwirl** [koreader/koreader#12165] [Ideas]
Hi, 

Is there anyone that can help me with this? I want to adjust my refresh to be triggered just a few miliseconds after pressing page turn button, instead of refreshing right away.

My device is Kobo Libra 2

I've delved into the lua code to make a userpatch or a plugin, but can't seem to find a way to do that (and I'm not a programmer, btw). 

Thank you.

**Comment by Frenzie** [comment]
Could you clarify what you're trying to do exactly? Any refreshes are triggered not on button press but after something like a new page has been pushed to the framebuffer.

(The strict answer to your question would be http://koreader.rocks/doc/modules/ui.uimanager.html#UIManager:scheduleIn but as implied above that's almost certainly not the right way to do things.)

**Reply by AryaTwirl** [reply]
Okay, what I mean by that is
I press turn page >> the display moves to another page >> after it finishes rendering another page, only then it triggers the refresh. 

So like partial refresh first but then full refresh right away, so to allow the display to finish rendering first before refreshing, not like immediately after pressing the button (or, as I have misunderstood, after the page pushed to the fb).

Someone has been developing a double refresh plugin. I have contacted that person, but no reply yet, and what I'm thinking is similar to his/hers (but without the first refresh). 

I got this idea from Boox Neo Reader "Depth" refresh mode. It exhibits the same behaviour as I have explained above, and it results in no ghosting on text content at all (still happening on pictures, but rendering pictures is completely another matter).

**Reply by Frenzie** [reply]
> I press turn page >> the display moves to another page >> after it finishes rendering another page, only then it triggers the refresh.

That simply describes how it works? Why do you think it works differently?

**Reply by AryaTwirl** [reply]
https://1drv.ms/v/s!AnaTLAnA3zWoiPxDI9rYNwHV5qmaDw?e=agjpSB

I have uploaded the video to demonstrate better. Let's just say when I manually refresh, that's when that full refresh but with delay happens. But instead of manually pressing, I want it to happen automatically with the similar amount of delay.

**Comment by poire-z** [comment]
May be `Gear > Screen > E-ink settings > Full refresh rate > Every page` is what you want ?

**Reply by AryaTwirl** [reply]
That's the thing, I always turn this feature on, but Libra 2 suffers from some slight ghosting compared to my Kindle. That's why I'm exploring about how to make the refresh delayed a bit.
